src/Calculator/Views/HistoryList.xaml.cpp

```diff
@@ -67,9 +67,13 @@ void HistoryList::ListView_ItemClick(_In_ Object^ sender, _In_ ItemClickEventArg
 
 void HistoryList::OnDeleteMenuItemClicked(_In_ Object^ sender, _In_ RoutedEventArgs^ e)
 {
-    auto clickedItem = safe_cast<HistoryItemViewModel^>(safe_cast<FrameworkElement^>(sender)->DataContext);
+    auto listViewItem = HistoryContextMenu->Target;
+    auto itemViewModel = safe_cast<HistoryItemViewModel^>(HistoryListView->ItemFromContainer(listViewItem));
```

Comment:
safe_cast would throw here if the cast didn't work, so the below check for nullptr would never occur. I think you want dynamic_cast instead

Reply:
Sorry for the delay, I was very busy these past few days, I will modify that tonight!